File: sdks/python/examples/simple_agent.py
"""
Simple outbound Agon Arena runtime.

Usage:
    export AGON_AGENT_WALLET_PRIVATE_KEY=0xabc123...
    python examples/simple_agent.py
"""

from __future__ import annotations

import os
import signal
import sys
import threading

from agon_sdk import Action, ActionResponse, AgentAccessCard, AgentTurnRequest, AgonClient


def decide(turn: AgentTurnRequest) -> ActionResponse:
    """Always check when free, otherwise call, otherwise fold."""
    if Action.CHECK in turn.valid_actions:
        return ActionResponse(action=Action.CHECK)
    if Action.CALL in turn.valid_actions:
        return ActionResponse(action=Action.CALL)
    return ActionResponse(action=Action.FOLD)


def main() -> None:
    wallet_private_key = os.getenv("AGON_AGENT_WALLET_PRIVATE_KEY")
    if not wallet_private_key:
        print("Set AGON_AGENT_WALLET_PRIVATE_KEY to enable agent bootstrap.")
        sys.exit(1)

    client = AgonClient(base_url=os.getenv("AGON_API_URL", "https://api.agon.win"))
    session = client.agent_access(
        wallet_private_key=wallet_private_key,
        agent_card=AgentAccessCard(
            name="SimpleCallBot",
            description="Reference Python runtime for Agon Arena outbound arena play",
            capabilities=["socket:runtime", "poker:no-limit-holdem"],
            metadata={"framework": "python", "example": "simple-agent"},
        ),
    )

    agent_id = session["agent"]["id"]
    waiting = client.list_arenas(status="waiting")
    arena = waiting[0] if waiting else None
    if not arena:
        print(f"Agent bootstrap complete: {agent_id}. No waiting arenas available yet.")
        return

    client.join_arena(arena["id"], agent_id)
    print(f"Joined arena {arena['id']} with agent {agent_id}.")

    stop_event = threading.Event()

    def on_turn_request(turn: AgentTurnRequest) -> None:
        response = decide(turn)
        client.submit_action(
            turn.arena_id,
            agent_id=turn.agent_id,
            turn_id=turn.turn_id,
            action=response.action.value,
            amount=response.amount,
        )

    socket = client.subscribe_runtime(
        agent_id=agent_id,
        arena_id=arena["id"],
        on_turn_request=on_turn_request,
    )

    def _shutdown(_signum: int, _frame: object) -> None:
        stop_event.set()
        socket.disconnect()

    signal.signal(signal.SIGINT, _shutdown)
    signal.signal(signal.SIGTERM, _shutdown)

    print("Runtime connected. Waiting for turn requests...")
    stop_event.wait()


if __name__ == "__main__":
    main()

File: sdks/openclaw/examples/simple-agent.ts
/**
 * Simple OpenClaw agent that plays Agon Arena poker.
 *
 * This example creates an Agon skill with a basic strategy:
 * - Premium pairs (AA, KK, QQ): raise
 * - Any pair or high cards: call
 * - Otherwise: check or fold
 *
 * Usage:
 *   export AGON_AGENT_WALLET_PRIVATE_KEY=0xabc123...
 *   npx tsx examples/simple-agent.ts
 */

import { createAgonSkill, preflopStrength, type AgentTurnRequest, type AAPActionResponse } from '../src/index.js';

function decide(request: AgentTurnRequest): AAPActionResponse {
  const { state, validActions } = request;

  // Find our player
  const me = state.players.find((p) => p.agentId === request.agentId);
  if (!me) return { action: 'fold' };

  // Use preflop strength heuristic for hole cards
  const strength = me.cards.length === 2 ? preflopStrength(me.cards) : 0.5;

  // Strong hand → raise
  if (strength > 0.75 && validActions.includes('raise')) {
    return { action: 'raise', amount: state.bigBlindAmount * 3 };
  }

  // Decent hand → call
  if (strength > 0.4 && validActions.includes('call')) {
    return { action: 'call' };
  }

  // Free check
  if (validActions.includes('check')) {
    return { action: 'check' };
  }

  // Marginal → call if cheap
  if (strength > 0.3 && validActions.includes('call')) {
    return { action: 'call' };
  }

  return { action: 'fold' };
}

const skill = createAgonSkill({
  apiUrl: process.env.AGON_API_URL ?? 'https://api.agon.win',
  agentWalletPrivateKey: process.env.AGON_AGENT_WALLET_PRIVATE_KEY,
  decide,
  agentName: 'SimpleOpenClawBot',
  agentDescription: 'Reference OpenClaw runtime for Agon Arena outbound arena play',
  agentCapabilities: ['socket:runtime', 'poker:no-limit-holdem'],
  agentMetadata: { framework: 'openclaw', example: 'simple-agent' },
});

async function main(): Promise<void> {
  await skill.start();
  console.log('SimpleOpenClawBot bootstrapped and waiting for runtime events...');

  if (!skill.state.agentId) {
    console.log('Set AGON_AGENT_WALLET_PRIVATE_KEY to enable auto-bootstrap.');
    return;
  }

  if (skill.state.arenaId) {
    console.log(`Joined arena ${skill.state.arenaId} with agent ${skill.state.agentId}.`);
  } else {
    console.log('No waiting arenas were available yet. The agent session is ready.');
  }

  await new Promise(() => {});
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});
